Assets/Scripts/Possession.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Possession : MonoBehaviour
+{
+
+    private Player player;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = gameObject.GetComponent<Player>();   
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    void OnCollisionStay(Collision collision)
+    {
+        if(collision.gameObject.tag == "Dead" && Input.GetAxis("Possess") != 0){
+            
+            Enemy deadEnemy = collision.gameObject.GetComponent<Enemy>();
+            player.MaxHealth = deadEnemy.MaxHealth;
+            player.Health = deadEnemy.MaxHealth;

```

Comment:
Be sure to invoke the player's health changed event, or else its HUD will not update.